### ISSUE TITLE
No button to test mailhost in Mail control panel

### ISSUE THREAD

**Issue by tiberiuichim** [plone/volto#3791]
There is a difference in functionality between the Plone Classic and Volto versions

**Comment by MdSahil-oss** [comment]
@tiberiuichim Would you like to elaborate the issue, I wanna work on this.

**Comment by tiberiuichim** [comment]
![image](https://user-images.githubusercontent.com/141568/198306259-034151e2-7e64-4374-bab7-9e50f5c958a3.png)

https://6-classic.demo.plone.org/@@mail-controlpanel

**Comment by MdSahil-oss** [comment]
@tiberiuichim Thanks! understood the issue. Soon will be raising a PR for this enhancement.

**Comment by MdSahil-oss** [comment]
@tiberiuichim does this look good ?

![Screenshot from 2022-10-31 18-50-55](https://user-images.githubusercontent.com/85174511/199023184-027b5813-e353-40aa-ac94-33798a41e10a.png)

**Comment by tiberiuichim** [comment]
@MdSahil-oss Yes, as far as I can say. You'll need to look at the plone.restapi implementation for the "save and send test email". Also, don't forget about the error messages.

**Comment by MdSahil-oss** [comment]
@tiberiuichim I spent some time on plone.restapi repository to find implementation of `Save and send test email` but couldn't find anything relevant, So thinking to implement a new method in plone.restapi repository which will "send a test email first and then trigger saving operation". what's your opinion on this.

**Comment by tiberiuichim** [comment]
I know that @avoinea was recently looking into the "mail sending options" of plone.restapi. Any pointers, on this, Alin?

**Comment by MdSahil-oss** [comment]
@avoinea Do you have any idea of it ?

**Comment by avoinea** [comment]
@MdSahil-oss 

1. There is an endpoint about sending arbitrary emails https://plonerestapi.readthedocs.io/en/latest/endpoints/email-send.html
2. And also for e-mail notifications https://plonerestapi.readthedocs.io/en/latest/endpoints/email-notification.html

Thus, I think we can use the second option for this feature request.

**Comment by MdSahil-oss** [comment]
@avoinea Are you sure ? because the logic that is implemented in the email-notification file is like:
```
data = json_body(self.request)

        sender_from_address = data.get("from", None)
        message = data.get("message", None)
        sender_fullname = data.get("name", "")
        subject = data.get("subject", "")

        if not sender_from_address or not message:
            raise BadRequest("Missing from or message parameters")

        overview_controlpanel = getMultiAdapter(
            (self.context, self.request), name="overview-controlpanel"
        )
        if overview_controlpanel.mailhost_warning():
            raise BadRequest("MailHost is not configured.")

        # Disable CSRF protection
        if "IDisableCSRFProtection" in dir(plone.protect.interfaces):
            alsoProvides(self.request, plone.protect.interfaces.IDisableCSRFProtection)

        contact_info_view = getMultiAdapter(
            (self.context, self.request), name="contact-info"
        )

        contact_info_view.send_message(
            dict(
                message=message,
                subject=subject,
                sender_from_address=sender_from_address,
                sender_fullname=sender_fullname,
            )
        )
```
because it seems like it's specialy built to send notifications when mail host is configured in the website, But as far as I understand the web site doesn't have any mailhost before the configuration so how it will send test email first and then save the given email as mailhost if test email sent successfully.

**Comment by MAX-786** [comment]
Hi there!
Does this issue needs love? I can see there's a linked PR but the issue is not yet resolved. Love to contribute if its needed. Thanks!

**Comment by acsr** [comment]
We stumbled today again over this issue over trying to activate autoregistration on a Volto site with no mailhost configured.
The registration link is not offered as UI element or link to an option in the login overlay.
When using the handcrafted url ./register with an unconfigured mailhost an error message "**Error** Unsuccessful HTTP response" is raised in the UI. But no error recorded in the default configuration to look after. (due to Unauthorized, NotFound, Redirect, Intercepted are disabled as events).

**Comment by davisagli** [comment]
> The registration link is not offered as UI element or link to an option in the login overlay.
When using the handcrafted url ./register with an unconfigured mailhost an error message "Error Unsuccessful HTTP response" is raised in the UI. But no error recorded in the default configuration to look after. (due to Unauthorized, NotFound, Redirect, Intercepted are disabled as events).

Those are not issues with the test button on the mailhost control panel. Please file each one separately (if they don't already exist). It's quite hard to keep track of what remains to be fixed if you report multiple issues together.

**Comment by acsr** [comment]
@davisagli sorry for mixing this up. I created a separate ticket related to the registration issues after deeper research. Nothing to do with the Mailhost setup. You still need the ClassicUI to test if the Mailhost config works.
